pennylane_lightning/lightning_gpu/_adjoint_jacobian.py

```diff
@@ -107,6 +107,9 @@ def _adjoint_jacobian_dtype(self):
         Returns: the AdjointJacobian class
         """
         if self._use_mpi:
+            if not MPI_SUPPORT:
+                warn(str(mpi_error), UserWarning)
+
```

Comment:
I move the warning message inside the MPI section to prevent showing it when LGPU was compilated without MPI support